File: sample/todo-app-ios/todo-app-ios/list/TodoListViewProxy.swift
//
//  TodoListViewImpl.swift
//  todo-app-ios
//
//  Created by Arkadii Ivanov on 22/02/2020.
//  Copyright © 2020 arkivanov. All rights reserved.
//

import Foundation
import TodoLib

class TodoListViewProxy: BaseMviView<TodoListViewModel, TodoListViewEvent>, TodoListView, ObservableObject {
    
    @Published var model: TodoListViewModel?
    
    override func render(model: TodoListViewModel) {
        self.model = model
    }
}

File: sample/todo-app-js/src/main/kotlin/list/TodoListViewProxy.kt
package list

import com.arkivanov.mvikotlin.core.view.BaseMviView
import com.arkivanov.mvikotlin.sample.todo.common.view.TodoListView

class TodoListViewProxy(
    private val updateState: (TodoListView.Model) -> Unit
) : BaseMviView<TodoListView.Model, TodoListView.Event>(), TodoListView {

    override fun render(model: TodoListView.Model) {
        updateState(model)
    }
}
